docs/style_guide.md

```diff
@@ -251,3 +251,27 @@ end
 
 `negate` is the version of the function that runs in OCaml, and `Checked.negate` is the one that runs inside of a SNARK circuit.
 
+### Polymorphic variants
+
+OCaml allows so-called polymorphic variants, with constructors marked with a backtick:
+
+```ocaml
+type truth = [ `True | `False ]
+```
+
+One advantage of such variants over ordinary algebraic datatypes is that they allow easy extensibility:
+
+```ocaml
+type uncertainty = [ truth | `Maybe ]
+```
+
+The constructors for polymorphic variants live in a global namespace, and they're not associated a unique defined type. One result is that the OCaml compiler can't check for exhaustive pattern-matching when using polymorphic variants. Their extensibility means both of these statements are correct:
+
+```ocaml
+let v1 : truth = `True
+let v2 : uncertainty = `True
+```
+
+There's an implied subtyping that may not be desired. In general, the typing of polymorphic variants is more complex than for ordinary datatype constructors.
+
+Unless there's need for extensibility, ordinary datatype constructors are usually preferred over polymorphic variants.
```

Comment:
What about anonymity? We use this a lot for anonymous convenience and for naming returns in/out of tuples. I'm not sure if I agree with this rule.

Reply:
I guess I disagree with those uses, because you could use records and defined types.

But maybe I'm just old-fashioned.